lib/rules/call-super-in-init.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const ember = require('../utils/ember');
+const utils = require('../utils/utils');
+
+//----------------------------------------------
+// General rule - Call _super in init lifecycle hooks
+//----------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforces super calls in init hooks',
+      category: 'General',
+      recommended: true
+    },
+    fixable: null, // or "code" or "whitespace"
+  },
+
+  create(context) {
+    const message = 'Call this._super(...arguments) in init hook';
+
+    const filePath = context.getFilename();
+
+    const report = function (node) {
+      context.report(node, message);
+    };
+
+    return {
+      CallExpression(node) {
+        if (!ember.isEmberComponent(node, filePath) &&
+            !ember.isEmberController(node, filePath) &&
+            !ember.isEmberRoute(node, filePath) &&
+            !ember.isEmberMixin(node, filePath)) return;
+
+        const initProperty = ember.getModuleProperties(node).find(property => property.key.name === 'init');
+
+        if (initProperty) {
+          const initPropertyBody = initProperty.value.body.body;
+          const fnExpressions = utils.findNodes(initPropertyBody, 'ExpressionStatement');
+
+          const hasSuper = fnExpressions.some((fnExpression) => {
+            const fnCallee = fnExpression.expression.callee;
```

Comment:
We should actually check if `fnExpression.expression` is `CallExpression` first.
This will work, but only due to additional check inside `isMemberExpression`, as the `fnCallee` in many cases will be `undefined`. Anyway we should not rely on the additional check inside `isMemberExpression` and be sure to pass the actual `CallExpression` there